crates/shared/src/subgraph.rs

```diff
@@ -59,14 +60,25 @@ impl SubgraphClient {
     where
         T: DeserializeOwned,
     {
-        self.client
-            .post(self.subgraph_url.clone())
-            .json(&Query { query, variables })
-            .send()
-            .await?
-            .json::<QueryResponse<T>>()
-            .await?
-            .into_result()
+        for _ in 0..MAX_NUMBER_OF_RETRIES {
+            match self
+                .client
+                .post(self.subgraph_url.clone())
+                .json(&Query {
+                    query,
+                    variables: variables.clone(),
+                })
+                .send()
+                .await?
+                .json::<QueryResponse<T>>()
+                .await?
+                .into_result()
+            {
+                Ok(result) => return Ok(result),
+                Err(err) => tracing::warn!("failed to query subgraph: {}", err),
+            }
+        }
+        Err(anyhow::anyhow!("failed to execute query on subgraph"))
```

Comment:
Looks good!

But would it make sense to match this error one level higher in order to not make the whole thread panic if the API is down? (sorry, I am missing a little bit context. Probably, its not needed)

Reply:
If I understand the code correctly, if the external source is down, we could not start our pod. I think this is an issue, as it could leave us in a panic loop. If it happens over night, its quite bad. I would rather prefer that it starts without the liquidity and throws an error. Wdyt?
Approved anyways, as this can be fixed in another PR